swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -371,38 +360,40 @@
       }()
     }
 
-    private func handleMenuVisibilityOrStatusChanged() {
-      let status = appStore.tunnelStore.status
-      if isMenuVisible && status == .connected {
-        appStore.tunnelStore.beginUpdatingResources()
+    private func resourceMenuTitle(_ newResources: [Resource]?) -> String {
+      guard let newResources = newResources else { return "Loading Resources..." }
+
+      if newResources.isEmpty {
+        return "No Resources"
       } else {
-        appStore.tunnelStore.endUpdatingResources()
+        return "Resources"
       }
     }
 
-    private func setOrderedResources(_ newOrderedResources: [DisplayableResources.Resource]) {
-      if resourcesTitleMenuItem.isHidden && resourcesSeparatorMenuItem.isHidden {
-        guard newOrderedResources.isEmpty else {
-          return
-        }
+    private func handleMenuVisibilityOrStatusChanged() {
+      if isMenuVisible && tunnelStore.status == .connected {
+        tunnelStore.beginUpdatingResources()
+      } else {
+        tunnelStore.endUpdatingResources()
```

Comment:
This part I don't quite get. Why do we have to begin updating resources? Does the tunnel process not push a list of resources to the GUI process when the connlib callback fires?

Also this looks familiar, I feel like another piece of code does the same thing. Either that or I'm remembering when I saw this piece in a commit diff this morning. Maybe one is for iOS and one is for macOS?

Reply:
Yeah the Menubar and Mainview are similar, but slightly different. It could probably be cleaned up with smarter SwiftUI code that handles both platforms.

The issue is that the tunnel process _can't_ push anything to the GUI process. There's no IPC mechanism for that, unless you get creative with daemons and stuff.

So `beginUpdatingResources` is a timer that's instantiated when the list of resources is visible and checks every 1s using a `sha256` of the list to determine if it needs to be sent over the IPC boundary.